### DISCUSSION TITLE
Uncaught SyntaxError: Invalid or unexpected token

### DISCUSSION THREAD

**Discussion by jonerickson** [livewire/livewire#7036] [1. Help]
`Uncaught SyntaxError: Invalid or unexpected token (at livewire.js?id=5d3e67e0:7804:1)`

Receiving the following JS error in the console when installing Livewire fresh. If I delete the NProgress comments at the end of the compiled Livewire file, the issue goes away. Anyone else experiencing this or have experienced this? 

Deleting the following at Line 7800 fixes the issue:

```
/* NProgress, (c) 2013, 2014 Rico Sta. Cruz - http://ricostacruz.com/nprogress
 * @license MIT */
/*! Bundled license information:

tabbable/dist/index.esm.js:
  (*!
  * tabbable 5.3.3
  * @license MIT, https://github.com/focus-trap/tabbable/blob/master/LICENSE
  *)

focus-trap/dist/focus-trap.esm.js:
  (*!
  * focus-trap 6.9.4
  * @license MIT, https://github.com/focus-trap/focus-trap/blob/master/LICENSE
  *)
*/
```

**Comment by jonerickson** [comment]
Finally found the issue, simply just an extra space at the top of one of my PHP files. The opening PHP tag started on line 2 instead of 1.

**Reply by PepinoCuellar** [reply]
What file? I've got the same problem, i can't find any solution

**Reply by jonerickson** [reply]
It was one of my routes file.

**Reply by untillnesss** [reply]
Hey, which file is this? I also have the same case

**Reply by TechDeptLunaSoftwareGroup** [reply]
Thanks for posting this... I was getting this error.  New to PHP had no idea about the whole buffer thing.  Back in the day when I was working with classic asp it defaulted to buffering the file.  Found out know it doesn't do that in php land.

But this saved me hours if not days of work.  With AI help I just had a script written to check all my php files to see if there was a problem.  I attached it if anyone else needs it.  It is written for ubuntu linux server.
[check-php-whitespace.sh](https://github.com/user-attachments/files/22145384/check-php-whitespace.sh)
Sure enough it found one.  fixed it and I was on my way.

**Comment by prykris** [comment]
I am still facing this issue. Anyone found a solution?

**Reply by prykris** [reply]
`php artisan vendor:publish --tag=livewire:assets --ansi --force` fixed it for me.

**Comment by titonova** [comment]
If it helps anyone, here is a temporary, tho admittedly bad idea , hack to fix the bug:

`Line 58 of \vendor\livewire\livewire\src\Drawer\Utils.php` add:

![image](https://github.com/user-attachments/assets/50075297-5d14-4d59-bb42-a6050b3677d3)


That's where the core problem is coming from afaik. This seems like it is a Livewire bug

**Reply by rizkihardinas** [reply]
you saved my life !

**Reply by titonova** [reply]
@rizkihardinas glad to hear that. But remember that any composer install or update operation may overwrite it . So keep that in mind
